ash/src/vk/prelude.rs

```diff
@@ -27,7 +29,7 @@ impl Packed24_8 {
     }
 }
 
-impl super::ColorComponentFlags {
-    /// Contraction of [`Self::R`] | [`Self::G`] | [`Self::B`] | [`Self::A`]
+impl vk::ColorComponentFlags {
+    /// Contraction of [`vk::ColorComponentFlags::R`] | [`vk::ColorComponentFlags::G`] | [`vk::ColorComponentFlags::B`] | [`vk::ColorComponentFlags::A`]
```

Comment:
This was throwing a very strange:

```rust
warning: unresolved link to `ColorComponentFlags::R`
  --> ash/src/vk/prelude.rs:31:26
   |
31 |     /// Contraction of [`Self::R`] | [`Self::G`] | [`Self::B`] | [`Self::A`]
   |                          ^^^^^^^ no item named `ColorComponentFlags` in scope
   |
   = note: `#[warn(rustdoc::broken_intra_doc_links)]` on by default
```

Despite residing in an `impl` block, `Self` should just expand to that. Maybe a `rustdoc` bug? Perhaps isolated to `const`.

Reply:
Weird! Reported upstream? I think this form of the doc is fine too, fortunately.

Reply:
https://github.com/rust-lang/rust/issues/93205

I planned to report it upstream but got discouraged after finding out that play.rust-lang.org _still_ can't build docs.

As it seems `Self` resolves to the type name only, not the full path. If I make sure that `ColorComponentFlags` is in scope (after all the error says ``no item named `ColorComponentFlags` in scope``) it'll work again.

What do you think? I've already gone back-and-forth multiple times on whether to write `Self` in public docs (makes for elegant links in markdown) or write out the full name. Do think I prefer `Self` though, that takes out the "was I looking at the `ColorComponentFlags` already?" mental question.

Reply:
Went with `Self::`, made sure that `ColorComponentFlags` is available, and linked to the Rust issue.

Reply:
I actually tend to favor dropping the prefix entirely (as if `Self::*` was in scope, though rust doesn't actually support globbing associated items (yet?)), but I don't have strong feelings; this is fine as is.

Reply:
We could us a "named (intradoc) link" (no idea if that's the technical terminology):

```markdown
Contraction of [`R`][Self::R] | [`G`][Self::G] | [`B`][Self::B] | [`A`][Self::A]
```

![image](https://user-images.githubusercontent.com/2325264/150653407-bedb018e-79d1-4339-b3db-c039ba2eef74.png)

Reply:
Yeah, exactly. Bit more concise to read, if a bit less convenient to write.